django/db/models/fields/__init__.py

```diff
@@ -843,18 +866,12 @@ def get_flatchoices(self, include_blank=True,
         first_choice = blank_choice if include_blank else []
         return first_choice + list(self.flatchoices)
 
-    def _get_val_from_obj(self, obj):
```

Comment:
Understood. Same goes for some of the sql.compiler refactoring too I'm guessing. 

It's going to end up a monolithic PR which will be difficult to review anyway, simply due to the nature of the changes -- I can't really avoid that.